snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/EffectiveTimeRestorer.java

```diff
@@ -0,0 +1,503 @@
+				final SnomedRelationship previousRelationship = (SnomedRelationship) previousVersion;
+						return relationshipToRestore.isActive() == previousRelationship.isActive()
+						&& relationshipToRestore.getModule().getId().equals(previousRelationship.getModuleId())
+						&& relationshipToRestore.getGroup() == previousRelationship.getGroup().intValue() 
+						&& relationshipToRestore.getUnionGroup() == previousRelationship.getUnionGroup().intValue() 
+						&& relationshipToRestore.getCharacteristicType().getId().equals(previousRelationship.getCharacteristicType().getConceptId())
+						&& relationshipToRestore.getModifier().getId().equals(previousRelationship.getModifier().getConceptId());	
+				
+			} else {
+				throw new UnexpectedTypeException("Unexpected component type '" + componentToRestore.getClass() + "'.");
+			}
+		} else if (componentToRestore instanceof SnomedRefSetMember && previousVersion instanceof SnomedReferenceSetMember) {
+			final SnomedReferenceSetMember previousMember = (SnomedReferenceSetMember) previousVersion;
+			if (componentToRestore instanceof SnomedAssociationRefSetMember) {
+				final SnomedAssociationRefSetMember memberToRestore = (SnomedAssociationRefSetMember) componentToRestore;
+				return memberToRestore.isActive() == previousMember.isActive()
+						&& memberToRestore.getModuleId().equals(previousMember.getModuleId()) 
+						&& memberToRestore.getTargetComponentId().equals(((SnomedConcept) previousMember.getProperties().get(SnomedRf2Headers.FIELD_TARGET_COMPONENT)).getId());
+			} else if(componentToRestore instanceof SnomedAttributeValueRefSetMember) {
+				final SnomedAttributeValueRefSetMember memberToRestore = (SnomedAttributeValueRefSetMember) componentToRestore;
+						return memberToRestore.isActive() == previousMember.isActive()
+						&& memberToRestore.getModuleId().equals(previousMember.getModuleId())
+						&& memberToRestore.getValueId().equals((String) previousMember.getProperties().get(SnomedRf2Headers.FIELD_VALUE_ID));
+			} else if (componentToRestore instanceof SnomedConcreteDataTypeRefSetMember) {
+				final SnomedConcreteDataTypeRefSetMember memberToRestore = (SnomedConcreteDataTypeRefSetMember) componentToRestore;
+				return memberToRestore.isActive() == previousMember.isActive() 
+						&& memberToRestore.getModuleId().equals(previousMember.getModuleId())
+						&& memberToRestore.getSerializedValue().equals(previousMember.getProperties().get(SnomedRf2Headers.FIELD_VALUE))
+						&& memberToRestore.getCharacteristicTypeId().equals(previousMember.getProperties().get(SnomedRf2Headers.FIELD_CHARACTERISTIC_TYPE_ID))
+						&& memberToRestore.getTypeId().equals(previousMember.getProperties().get(SnomedRf2Headers.FIELD_TYPE_ID))
+						&& memberToRestore.getGroup() == ((Integer )previousMember.getProperties().get(SnomedRf2Headers.FIELD_RELATIONSHIP_GROUP)).intValue();
+			} else if(componentToRestore instanceof SnomedDescriptionTypeRefSetMember) {
```

Comment:
Fix spacing, please! :)

Reply:
This doesn't look the same inside my IDE.